Helper/Entity/ProductHelper.php

```diff
@@ -502,6 +502,15 @@ public function setSettings($indexName, $indexNameTmp, $storeId, $saveToTmpIndic
         // $this->deleteUnusedReplicas($indexName, $replicas, $setReplicasTaskId);
 
         if ($saveToTmpIndicesToo === true) {
+            try {
+                $this->algoliaHelper->copySynonyms($indexName, $indexNameTmp);
+                $this->logger->log('
+                    Copying synonyms from production index to TMP one to not to erase them with the index move.
```

Comment:
Copying synonyms from production index to TMP one to not erase them with the index move.